### ISSUE TITLE
`sendMessageStream` with Chinese prompt returns malfunctioning chunks

### ISSUE THREAD

**Issue by Fndroid** [google/generative-ai-js#9]
## Expected Behavior
`sendMessageStream` with Chinese prompt should returns right chunks.

## Actual Behavior
`sendMessageStream` with Chinese prompt returns malfunctioning chunks which contains `���`.

## Steps to Reproduce the Problem

1.  
    ```js
    const prompt = '创作一首古诗';
    const chat = await model.startChat(history);
    const result = await chat.sendMessageStream(prompt);
    let text = ''
    for await (const chunk of result.stream) {
      text += chunk.text();
    }
    console.log(text)
    ```
1. get result like this:
   ```
    **寒夜**
    
    月寒山冷夜漫长，
    秋声���瑟送寒凉。
    银辉霜雪遍地洒，
    红叶舞尽染霜霜。
    
    寒风凛冽凌人骨，
    万籁俱寂月霜寒。
    古刹钟声伴孤影，
    一曲悲歌满心酸。
    
    寒江衰柳起愁思，
    霜梅雪月亦难禁。
    孤篇残卷叹今昔，
    满目疮痍叹世事。
    
    寒������滚������来，
    哀鸿遍野亦不悔。
    纵然前路多险阻，
    岂能轻言半途弃？
   ```
1. text contains `���`

## Specifications

- Version: ^0.1.1
- Platform: macOS

**Comment by hsubox76** [comment]
We currently handle each chunk coming in by parsing it to utf-8 which isn't always appropriate for certain characters that may be more bytes. We need to parse it in a way that can handle characters of different byte lengths.

**Comment by Ark-kun** [comment]
Maybe the service should not split chunks in the middle of a character.

**Comment by Fndroid** [comment]
I've tested the latest build, but the issue still persists.

It appears that the `text` in the response stream is incorrect.

<img width="993" alt="image" src="https://github.com/google/generative-ai-js/assets/16091562/1f028a83-127a-4575-a733-cc6864034c95">

**Comment by alx13** [comment]
Confirmed.
Should be fixed with https://github.com/google/generative-ai-js/pull/22 :

```
import { GoogleGenerativeAI } from "@google/generative-ai";

async function run() {
  const model = new GoogleGenerativeAI(
    process.env["GOOGLE_API_KEY"] as string
  ).getGenerativeModel({
    model: "gemini-pro",
  });

  const prompt = "创作一首古诗";
  const chat = await model.startChat();
  const result = await chat.sendMessageStream(prompt);
  let text = "";
  for await (const chunk of result.stream) {
    text += chunk.text();
  }
  console.log(text);
}

if (require.main === module) {
  run().catch((e) => {
    console.error(e);
    process.exit(1);
  });
}
```
./node_modules/.bin/ts-node sdk.ts                           
山涧飞流
潺潺水声
悠然入耳
沁入心田

清风徐来
拂动柳枝
摇曳生姿
婆娑起舞

花红柳绿
莺歌燕舞
百鸟争鸣
欢欣雀跃

山清水秀
风景如画
如诗如画
令人陶醉
```

```

**Comment by hsubox76** [comment]
Fix now published in 0.1.3, let us know if it's still an issue in 0.1.3 and we will reopen.

**Comment by coderyqy** [comment]
model `geminiAI-vision`  `generateContentStream` with Chinese prompt returns malfunctioning chunks which contains ������

const result = await model.generateContentStream([
        '用中文回答',
        ...history,
        {
          inlineData: {
            data: data.base64,
            mimeType: data.mimeType,
          },
        },
      ]);


 宫保鸡丁是一道著名的川菜，起源于清朝山东巡抚丁宝桢的家厨，以花生、黄瓜、胡萝卜
、������、���丝等配料，再佐以干辣椒、花椒等调味料炒制而成。

 这碟粉真的太好吃了！我从来没有吃过这么好吃的粉。粉很滑，很入味，而且配菜也很丰富。我
���定会再来吃的！
